### PR TITLE
[CPU]: Added ROIAlignRotated basic impl.

### PR DESCRIPTION
**Pull request by pkowalc1**
### Details:
 - Added basic implementation of ROIAlignRotated
 - It uses implementation from core::reference - the goal was just to add support for the op, not the optimized implementation.
 - No unit tests added, since impl is already tested by core::reference and functional tests.

### Tickets:
 - [CVS-135847](https://jira.devtools.intel.com/browse/CVS-135847)

### CODE REVIEW

**Review by praasz — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                getSrcMemoryAtPort(2)->getPrecision(),
+                ov::element::i64,
+                batch_indices_size);
+
+    ov::reference::roi_align<T, ov::reference::roi_policy::ROIAlignRotatedOpDefPolicy>(
+        getSrcDataAtPortAs<const T>(0),
+        getSrcDataAtPortAs<const T>(1),
+        batch_indices_vec_scaled_up.data(),
+        getDstDataAtPortAs<T>(0),
+        getSrcMemoryAtPort(0)->getShape().toPartialShape().to_shape(),  //< Conversion from intel_cpu::Shape to
```
[the remark this answers is not in the document]
Comment (pkowalc1) [reply src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:74]: Fixed.

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+#include "common/cpu_convert.h"
+#include "openvino/reference/roi_align.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+ROIAlignRotated::ROIAlignRotated(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
+    std::string errorMessage;
```
[the remark this answers is not in the document]
Comment (pkowalc1) [reply src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:18]: Fixed.

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "roi_align_rotated.h"
+
+#include <openvino/opsets/opset14.hpp>
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:7]: Please include just required operation headers.

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+#include "common/cpu_convert.h"
+#include "openvino/reference/roi_align.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:15]: Need to add fn `isSupportedOperation()`, like in other CPU nodes

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+void ROIAlignRotated::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    ov::element::Type inputPrec0 = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type outputPrec = getOriginalOutputPrecisionAtPort(0);
+
+    addSupportedPrimDesc(
+        {{LayoutType::ncsp, inputPrec0}, {LayoutType::ncsp, ov::element::f32}, {LayoutType::ncsp, ov::element::i32}},
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:38]: As per specification, 1st and 2nd inputs have the same precisions. Why the second one is always f32?
Also need to check this value. Plugin supports only f32, f16 and bf16 floating point types.

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        samplingRatio,
+        spatialScale,
+        ov::op::v3::ROIAlign::PoolingMode::AVG,
+        ov::op::v9::ROIAlign::AlignedMode::ASYMMETRIC,
+        clockwiseMode);
+}
+
+void ROIAlignRotated::execute(dnnl::stream) {
+    const ov::element::Type type = getOriginalInputPrecisionAtPort(0);
+    executeImpl<ov::element::f32>();
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:88]: Why this fn is called twice?

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+#define CASE(OV_TYPE)                        \
+    case ov::element::OV_TYPE:               \
+        executeImpl<ov::element::OV_TYPE>(); \
+        break;
+
+    switch (type) {
+        CASE(bf16);
+        CASE(f16);
+        CASE(f32);
+        CASE(f64);
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:99]: Not supported by plugin

**Comment by nshchego** [comment]
Please add CPU layer tests

File: src/plugins/intel_cpu/CMakeLists.txt [hunk 8]
```diff
@@ -186,6 +186,11 @@ if (ENABLE_SNIPPETS_LIBXSMM_TPP)
     target_include_directories(${TARGET_NAME} SYSTEM PRIVATE $<TARGET_PROPERTY:xsmm,INCLUDE_DIRECTORIES>)
 endif ()
 target_include_directories(${TARGET_NAME} SYSTEM PRIVATE $<TARGET_PROPERTY:dnnl,INCLUDE_DIRECTORIES>)
+
+# Temporal solution to use template reference implementations in cases where optimizied implementation
```
Comment (nshchego) [inline src/plugins/intel_cpu/CMakeLists.txt:190]: Please add a ticket number

Comment (mlukasze) [reply src/plugins/intel_cpu/CMakeLists.txt:190]: CVS-141656

File: src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+void ROIAlignRotated::execute(dnnl::stream) {
+    const ov::element::Type type = getOriginalInputPrecisionAtPort(0);
+    executeImpl<ov::element::f32>();
+
+#define CASE(OV_TYPE)                        \
+    case ov::element::OV_TYPE:               \
+        executeImpl<ov::element::OV_TYPE>(); \
+        break;
+
+    switch (type) {
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp:95]: Use OV_SWITCH instead

**Pull request merged**